keps/sig-node/3953-node-resource-hot-plug/README.md

```diff
@@ -0,0 +1,797 @@
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+Not relevant, As this kubelet specific feature and does not impact other components.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name:NodeResourceHotPlug
+    - Components depending on the feature gate: kubelet
+- [ ] Other
+    - Describe the mechanism:
+    - Will enabling / disabling the feature require downtime of the control
+      plane?
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node?
+
+###### Does enabling the feature change any default behavior?
+
+No. This feature is guarded by a feature gate. Existing default behavior does not change if the
+feature is not used. 
+Even if the feature is enabled via feature gate, If there is no change in 
+node configuration the system will continue to work in the same way.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Once disabled any hot plug of resources won't reflect at the cluster level. 
```

Comment:
Please describe this process.  I think it requires restarting at least the kubelet?  When the kubelet restarts, can it be in a state where pods are running that wouldn't be running otherwise?  What does the kubelet do about this?

Reply:
sure, I will update and we need to turn off the feature gate and restart the kubelet.
also could you please help me understand on what reason you are suspecting pods to turn into running from pending post kubelet restart?